### ISSUE TITLE
File is not saved, error is thrown

### ISSUE THREAD

**Issue by Antonytm** [TeeSeal/coub-dl#17]
**Steps to reproduce:**
1. Run coub-dl -i https://coub.com/view/1oag7h -o 1oag7h.mp4 --loop 10 --time 12 -d

```
Getting error:
[tls @ 000001cd70cf88c0]
Error in the pull function.

https://coubsecure-s.akamaihd.net/get/b174/p/coub/simple/cw_looped_audio/4b8588de03f/1c6cc2291842ed3987051/high_1550172977_high.mp3: I/O error
```

**Comment by TeeSeal** [comment]
Hi,

Thanks for the report but unfortunately I am unable to reproduce this 😕Sounds like a network/TLS issue. 
Does it happen for every coub you try to download or only this one in particular?

**Comment by Antonytm** [comment]
It is almost for all. About 8 from 10 execution of coub-dl fails.

I can open [mp3 link](https://coubsecure-s.akamaihd.net/get/b174/p/coub/simple/cw_looped_audio/4b8588de03f/1c6cc2291842ed3987051/high_1550172977_high.mp3) in browser, that is why I think that problem is not in network/TLS.

**Comment by TeeSeal** [comment]
Can you try running:
```
ffmpeg -i https://coubsecure-s.akamaihd.net/get/b174/p/coub/simple/cw_looped_audio/4b8588de03f/1c6cc2291842ed3987051/high_1550172977_high.mp3 song.mp3
```
And let me know how that goes.

**Comment by Antonytm** [comment]
![image](https://user-images.githubusercontent.com/647813/73815760-9d4c4a80-47ef-11ea-98af-fec346af3308.png)

**Comment by Antonytm** [comment]
but URL works in browser...

**Comment by TeeSeal** [comment]
Yeah that looks a problem between you and ffmpeg 🤔 

I found some clues online https://stackoverflow.com/questions/54139166/python-youtube-ffmpeg-session-has-been-invalidated

I could try downloading the audio before trying to build the coub file, though it's going to be difficult for me to tell if it fixes the issue since I can't reproduce it in the first place. The downside to that will be that the whole process will become slower (because I have to download the audio instead of streaming it straight from the source). Maybe I'll put this functionality behind a flag or something.

Will let you know when I've pushed the update.
